File: Leetcode/2900 - Longest Unequal Adjacent Groups Subsequence I/solution.py
def getLongestSubsequence(self, words: List[str], groups: List[int]) -> List[str]:
    alt_seq_0: List[str] = []
    alt_seq_1: List[str] = []
    expect_0: int = 0
    expect_1: int = 1

    for i in range(len(groups)):
        if groups[i] == expect_0:
            alt_seq_0.append(words[i])
            expect_0 = 1 - expect_0
        if groups[i] == expect_1:
            alt_seq_1.append(words[i])
            expect_1 = 1 - expect_1

    return alt_seq_0 if len(alt_seq_0) >= len(alt_seq_1) else alt_seq_1

File: Leetcode/2900 - Longest Unequal Adjacent Groups Subsequence I/solution.ts
function getLongestSubsequence(words: string[], groups: number[]): string[] {
    const altSeq0: string[] = [];
    const altSeq1: string[] = [];
    let expect0: number = 0;
    let expect1: number = 1;

    for (let i: number = 0; i < groups.length; i++) {
        if (groups[i] === expect0) {
            altSeq0.push(words[i]);
            expect0 = 1 - expect0;
        }
        if (groups[i] === expect1) {
            altSeq1.push(words[i]);
            expect1 = 1 - expect1;
        }
    }

    return (altSeq0.length >= altSeq1.length) ? altSeq0 : altSeq1;
};